test/unit/api/compress/compress_test_filter.hpp

```diff
@@ -0,0 +1,45 @@
+
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+**/
+//==================================================================================================
+#pragma once
+#include "test.hpp"
+template<typename Type> struct rvv_compress_filter
+{
+  static consteval bool is_enabled()
+  {
+    // For vlen = 128, we test types with all integer up to lmul==1
+    constexpr auto bit_size = 128;
+    using scalar_t          = typename Type::value_type;
+    if( sizeof(Type) <= bit_size / 8 && std::is_integral_v<scalar_t> ) return true;
+    constexpr auto cardinal          = eve::cardinal_v<Type>;
+    constexpr auto expected_cardinal = eve::expected_cardinal_v<scalar_t>;
+    // and double
+    if( sizeof(scalar_t) != 8 || !std::is_floating_point_v<scalar_t> ) return false;
+    // for expected cardinal (lmul==8)
+    if( expected_cardinal == cardinal ) return true;
+    // for combined type
+    if( 2 * expected_cardinal == cardinal ) return true;
+    return false;
+  }
```

Comment:
I find this function way too difficult to parse through. What's bit_size / 8.

sizeof(Type) <= bit_size / 8 should be true for all types. What are you trying to do. 

If you explain in plain english what you want, we can do figure out how to write it.

Reply:
maximum register of 16 bytes is not good enough. It means you never test any multiple bigger than 1. 

What is your problem? compiler running out of memory?

Reply:
We already have a very large filtering on types.
Because not all of the interesting types are in the pool, it is very hard to judge what's tested. 

What if we remove just some of the signs?

Smth like:

```
return eve::detail::is_one_of<eve::element_type_t<T>>(
 types<std::int8_t, std::uint16_t, float, double>{}
);
```

is one of function is defined here: https://github.com/jfalcou/eve/blob/11c8d8c8ee2c9b42fa3a25613d6f5bc1efece991/include/eve/detail/meta.hpp#L75

I feel that in order to understand what you are doing here otherwise, you need to go into test and see what's removed there.

Reply:
You mean in compress filter try to specify exact types, instead of reducing fully? Yep, let's try. You suggest to check `std::int8_t, std::uint16_t, float, double` only?

Reply:
> `exact types, instead of reducing fully`? 

What do you mean by reducing fully? I'm just looking for some filtering that make more sense.

Reply:
Please take a look now. Did I understand you right?